docs/decisions/0039-agent-interactions.md

```diff
@@ -0,0 +1,460 @@
+---
+# These are optional elements. Feel free to remove any of them.
+status: proposed
+contact: crickman
+date: 2024-01-24
+deciders: semenshi, matthewbolanos, markwallace-microsoft,alliscode
+informed:
+---
+
+# SK Agent Interactions and Architecture
+
+## Context and Problem Statement
+- Agent abstraction must adapt to different modalities
```

Comment:
"Agent abstraction must adapt to different modalities e.g. Chat Completion, OpenAI Assistants <sup>1</sup>"

and then below where the detailed descriptions of these things live also include the <sup>1</sup>

For someone who is new to this I think this will help bring clarity